### PR TITLE
[tokenizer] [tokenizer] Adds batch encoding support for question answering

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java [hunk 1]
```diff
@@ -40,49 +43,71 @@ public class QaServingTranslator implements Translator<Input, Output> {
...
-    public Batchifier getBatchifier() {
-        return translator.getBatchifier();
+        this.batchTranslator = translator.toBatchTranslator();
     }
 
     /** {@inheritDoc} */
     @Override
     public void prepare(TranslatorContext ctx) throws Exception {
         translator.prepare(ctx);
+        translator.prepare(ctx);
```
Comment (siddvenk) [inline api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java:53]: duplicate line?

File: api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java [hunk 2]
```diff
@@ -40,49 +43,71 @@ public class QaServingTranslator implements Translator<Input, Output> {
...
+                    return batchTranslator.processInput(ctx, inputs);
+                }
+
+                qa = JsonUtils.GSON.fromJson(json, QAInput.class);
             } catch (JsonParseException e) {
                 throw new TranslateException("Input is not a valid json.", e);
             }
+        } else if (content.contains("question") && content.contains("paragraph")) {
+            String question = input.getAsString("question");
+            String paragraph = input.getAsString("paragraph");
```
Comment (siddvenk) [inline api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java:82]: Is "paragraph" the key we want to enforce here? It seems like in huggingface for QA they expect "context" https://huggingface.co/docs/transformers/tasks/question_answering#inference.

File: api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java (hunk 1 above)
Comment (frankfliu) [reply api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java]: copy paste error, fixed

File: api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java (hunk 2 above)
Comment (frankfliu) [reply api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java]: We use paragraph for a long time, I made changes to compatible with huggingface

Comment (siddvenk) [reply api/src/main/java/ai/djl/modality/nlp/translator/QaServingTranslator.java]: Got it - I wasn't sure if it was compatible

**Review by siddvenk — APPROVED** [review]

**Pull request merged**